src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java

```diff
@@ -358,7 +360,10 @@ public void openSearch(PlayerProfile profile, String input, boolean addToHistory
                 break;
             }
 
-            if (!slimefunItem.isHidden() && isSearchFilterApplicable(slimefunItem, searchTerm)) {
+            if (!slimefunItem.isHidden()
+                    && (showHiddenItemInSearch || !slimefunItem.getItemGroup().isHidden(p))
+                    && isSearchFilterApplicable(slimefunItem, searchTerm)
+            ) {
```

Comment:
I hate it when people format if-statements this way.
Can you perhaps make a dedicated method for this check to simplify this?